### PR TITLE
scale/zoom RPM lights to upper RPM band

### PR DESCRIPTION
**Pull request by Oinosseus**
- lowest RPM value adjustable by 'minrpmrel' parameter in settings.ini [in percent]
- also using round() to display last RPM light before reaching exactly 100% maxRPM

### CODE REVIEW

**Comment by dchoptiany** [comment]
Hello,
thank you for your contribution. I need some time to think about it, but  I'll definitely come back to this pull request.

File: LmpGUI.py [hunk 1]
```diff
@@ -149,9 +152,13 @@ def getSpeed():
...
 def drawRPMLights():
-	global scale, maxRPM
+	global scale, maxRPM, minRPMRel
 
-	numberOfLights = int(ac.getCarState(0, acsys.CS.RPM) * 12 / maxRPM)
+	minRPMNorm = minRPMRel / 100.0
+	currentRPM = ac.getCarState(0, acsys.CS.RPM)
+	numberOfLights = round(12 * ((currentRPM / maxRPM) - minRPMNorm) / (1 - minRPMNorm))
+	if numberOfLights < 0:
+		numberOfLights = 0
```
Comment (dchoptiany) [inline LmpGUI.py]: I don't like these two lines. Maybe there is a reason I don't understand, but this seems unnecessary. Could you also shortly describe the idea behind this pull request and minRPMRel value? :)

**Comment by Oinosseus** [comment]
At the moment your lights show the RPM in a range from 0% to 100%. Often drivers don't care for low RPMs and want more resolution for higher RPMs. With this patch a driver can set the minimum RPM to example 30%. So, the lights go from 30% to 100%.

Since the formula is a linear rescaling, numberOfLights will be negative when the current RPM is lower than the minimum RPM. So, the two mentioned lines limit the lower range of numberOfLights to zero.

My implementation does read the minimum RPM value from the settings.ini file. This can be improved later by adding a GUI input element.

**Comment by dchoptiany** [comment]
Thank you for explaining your idea. I think it makes sense now :) I will resolve conflicts and add GUI input for users to easily set lower bound in percent. Thanks for contribution :)

**Pull request merged**